### PR TITLE
Fix relation local save

### PR DESCRIPTION
**Pull request by webrouse**
Part of: https://keboola.atlassian.net/browse/COM-1137

### CODE REVIEW

File: internal/pkg/diff/diff.go [hunk 1]
```diff
@@ -168,10 +168,6 @@ func (d *Differ) newOptions() cmp.Options {
 		cmpopts.AcyclicTransformer("strByLine", func(s string) []string {
 			return strings.Split(s, "\n")
 		}),
```
Comment (webrouse) [inline internal/pkg/diff/diff.go:170]: Nie je tu toho vela. 3 fixy v kode -> prvy commit.
Druhy commit je novy funkcionalny test.

File: internal/pkg/diff/diff.go [hunk 2]
```diff
@@ -168,10 +168,6 @@ func (d *Differ) newOptions() cmp.Options {
...
-		// Compare only relations defined on the API side
```
Comment (webrouse) [inline internal/pkg/diff/diff.go:171]: Tu som sa trocha sekol.
Diff sa pouziva aj pri ukladani `local` stavu, a preto tam musia byt vsetky `relations`, nie iba API side, ale aj manifest side.

Ak by som to nezmazal, ... tak ak pri `pull` dostane nejaky existujuci objekt novu `Relation`, tak sa neulozi do manifestu, pretoze objekt nie je v diffe.

File: internal/pkg/diff/diff_test.go [hunk 3]
```diff
@@ -329,7 +327,9 @@ func TestDiffRelations(t *testing.T) {
 	differences := differ.diffValues(objectState.Key(), rObject.Relations, lObject.Relations)
 	expected := `
   - api side relation "path/to/target"
+  - manifest side relation mocked key "foo"
```
Comment (webrouse) [inline internal/pkg/diff/diff_test.go:330]: Upraveny test k ^^^^.

File: internal/pkg/remote/manager.go [hunk 4]
```diff
@@ -226,7 +226,7 @@ func (u *UnitOfWork) Invoke() error {
 	// Generate local path if needed
 	pathsUpdater := u.localManager.NewPathsGenerator(false)
 	for _, objectState := range u.newObjectStates {
-		if !objectState.HasLocalState() {
+		if objectState.GetObjectPath() == "" {
```
Comment (webrouse) [inline internal/pkg/remote/manager.go:229]: Blba podmienka:
- najprv sa vzdy nacitava `remote` state:
https://github.com/keboola/keboola-as-code/blob/624fb90a2e336295924bc6da453724a0d2ac0d01/internal/pkg/state/state.go#L70-L86
- takze tu v `remote.Manager` bude `objectState.HasLocalState` vzdy `false`.
- no manifest objektu je uz nacitany (ak existuje) a objekt uz ma `path`.
- preto opravena podmienka, a pytam sa priamo na `path`.
- ak objekt existuje iba `remote`, tak nebude mat `path` a ta sa musi vygenerovat.

File: test/pull-variables-add-relation/expected-stdout [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Plan for "pull" operation:
+  CH C my-branch/extractor/ex-generic-v2/with-variables-1 | changed: Relations
+  CH C my-branch/other/keboola.variables/variables | changed: Relations
+  CH R my-branch/other/keboola.variables/variables/values/default | changed: Relations
+Plan for "rename" operation:
+  - my-branch -> main
+  - main/{other/keboola.variables/variables -> extractor/ex-generic-v2/with-variables-1/variables}
```
Comment (webrouse) [inline test/pull-variables-add-relation/expected-stdout:7]: Novy test:
- config + variables config existuju `lokalne` (aj `remote`)
- ale lokalne nie su prepojene
- `pull` prida `relations` do `manifest.json` a presunie `variables` do `config`.
- trocha edge case, ale ked funguje toto, tak funguju aj jednoduchsie priklady (uz pokryte testami).

File: test/pull-variables-add-relation/out/.keboola/manifest.json [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      "id": "%%TEST_BRANCH_MAIN_CONFIG_WITH_VARIABLES_1_ID%%",
+      "path": "extractor/ex-generic-v2/with-variables-1",
+      "rows": []
+    },
+    {
+      "branchId": %%TEST_BRANCH_MAIN_ID%%,
+      "componentId": "keboola.variables",
+      "id": "%%TEST_BRANCH_MAIN_CONFIG_VARIABLES_ID%%",
+      "path": "variables",
+      "relations": [
```
Comment (webrouse) [inline test/pull-variables-add-relation/out/.keboola/manifest.json:40]: V `out` su pridane `relations`.

File: test/pull-variables-add-relation/in/.keboola/manifest.json [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      "path": "my-branch"
+    }
+  ],
+  "configurations": [
+    {
+      "branchId": %%TEST_BRANCH_MAIN_ID%%,
+      "componentId": "ex-generic-v2",
+      "id": "%%TEST_BRANCH_MAIN_CONFIG_WITH_VARIABLES_1_ID%%",
+      "path": "extractor/ex-generic-v2/with-variables-1",
+      "rows": []
```
Comment (webrouse) [inline test/pull-variables-add-relation/in/.keboola/manifest.json:33]: V `in` relations nie su.

**Review by JakubMatejka — APPROVED** [review]
